zaza/openstack/utilities/openstack.py

```diff
@@ -483,8 +484,68 @@ def get_admin_net(neutron_client):
             return net
 
 
+def _plumb_guest_nic(server_name, mac_address, dvr_mode=None):
+    """In guest server_name, add port with mac_address to netplan.
+
+    :param server_name: Hostname of instance
+    :type server_name: string
+    :param mac_address: mac address of nic to be added to netplan
+    :type mac_address: string
+    :param dvr_mode: Using DVR mode or not
+    :type dvr_mode: boolean
+    """
+    if dvr_mode:
+        application_name = 'neutron-openvswitch'
+    else:
+        application_name = 'neutron-gateway'
+
+    unit_name = juju_utils.get_unit_name_from_host_name(
+        server_name, application_name)
+
+    run_cmd_nic = "ip addr|grep {} -B1".format(mac_address)
+    interface = model.run_on_unit(unit_name, run_cmd_nic)
+    interface = interface['Stdout'].split(' ')[1].replace(':', '')
+
+    run_cmd_netplan = """sudo egrep -iR '{}|{}' /etc/netplan/
+                        """.format(mac_address, interface)
+
+    netplancfg = model.run_on_unit(unit_name, run_cmd_netplan)
+
+    if (mac_address in str(netplancfg)) or (interface in str(netplancfg)):
+        logging.warn("mac address {} or nic {} already exists in "
+                     "/etc/netplan".format(mac_address, interface))
+        return
+    body_value = textwrap.dedent("""\
+        network:
+            ethernets:
+                {0}:
+                    dhcp4: false
+                    dhcp6: true
+                    optional: true
+                    match:
+                        macaddress: {1}
+                    set-name: {0}
+            version: 2
+    """.format(interface, mac_address))
+    logging.debug("plumb guest interface debug info:")
+    logging.debug("body_value: {}\nunit_name: {}\ninterface: {}\nmac_address:"
+                  "{}\nserver_name: {}".format(body_value, unit_name,
+                                               interface, mac_address,
+                                               server_name))
+    netplan_file = open("60-dataport.yaml", "w")
```

Comment:
I think its worth using a context manager to write a temporary file here, then it looks after creating it safely and the cleanup.